### DISCUSSION TITLE
Long waiting queue

### DISCUSSION THREAD

**Discussion by ghost** [ROM-builders/temporary#23416] [Q&A]
@Apon77 I know that cirrus ci is providing us free servers for building roms and that's too awesome! Loved the work and I am praising your and cirrus's hardwork to help builders like us. But this may seem harsh but it is true that pr users are not getting enough time to build roms. Here are the following reasons:-
a) If approval isn't required then there is no guarantee that build will succeed and in case they face an error, by the time approval is required for triggering builds and they will have to wait for few hours, sometimes even for days!
b) Seems that push users are the only ones who are taking complete benifit of the service.
c) I know that there are some buildbots who are abusing the system, but there are needy people who "really" need some time to build on the platform
d) I even know that push users are experienced and are having a proper knowledge about development, but there are some push users who trigger 3-4 builds in a day. And most probably, the system is getting abused in that scenario too.
e) And the worst case becomes when approval required issue persists for 2-3 days. In this case pr users are troubled as they don't have any other means to build.
I really appreciate the rules laid down by the organisation, but as pr users, we also need time to cope up with our builds. So it's my request on behalf of all pr users to either decrease the approval time or to separate the pr users and push users. Like pr users will build on one server and push users on another. I hope that some action will be taken regarding this issue. Hoping for a positive response

**Comment by Apon77** [comment]
I increased the limit from 2h to 3h, it will do some help I think. 
I think that was the starting of the month when sp were released for all ROMs, thats why pressure was high. Because nowadays pressure is low on the server. 
Also, I published a website http://cirrus-ci.000.pe where you can see who built how many times by using the filter button. Hope it will help.
Thank you.

**Reply by ghost** [reply]
So sir, the 2 hour limit which cirrus gives to the free users has been increased to 3 hours ?

**Reply by Apon77** [reply]
Yes
